pyamg/classical/interpolate.py

```diff
@@ -38,34 +38,208 @@ def direct_interpolation(A, C, splitting):
      [0.  1.  0. ]
      [0.  0.5 0.5]
      [0.  0.  1. ]]
-
     """
     if not isspmatrix_csr(A):
         raise TypeError('expected csr_matrix for A')
 
     if not isspmatrix_csr(C):
         raise TypeError('expected csr_matrix for C')
+    
+    if theta is not None:
+        C = classical_strength_of_connection(A, theta=theta, norm=norm)
+    C.eliminate_zeros()
+
+    # Interpolation weights are computed based on entries in A, but subject to the
+    # sparsity pattern of C.  So, copy the entries of A into sparsity pattern of C.
+    C.data[:] = 1.0
+    C = C.multiply(A)
+
+    P_indptr = np.empty_like(A.indptr)
+    amg_core.rs_direct_interpolation_pass1(A.shape[0], C.indptr, C.indices, 
+                                           splitting, P_indptr)
+    nnz = P_indptr[-1]
+    P_colinds = np.empty(nnz, dtype=P_indptr.dtype)
+    P_data = np.empty(nnz, dtype=A.dtype)
+
+    amg_core.rs_direct_interpolation_pass2(A.shape[0], A.indptr, A.indices,
+                                           A.data, C.indptr, C.indices, C.data,
+                                           splitting, P_indptr, P_colinds, P_data)
+
+    nc = np.sum(splitting)
+    n = A.shape[0]
+    return csr_matrix((P_data, P_colinds, P_indptr), shape=[n,nc])
+
+
+def standard_interpolation(A, C, splitting, theta=None, norm='min', modified=True):
```

Comment:
@bensworth @lukeolson  Should we rename this "standard_classical_interpolation",  "std_classical_interpolation",  or just "classical_interpolation"

The paper https://onlinelibrary.wiley.com/doi/abs/10.1002/nla.559 
calls something else "standard" interpolation, and I think this could be confusing.  I'm not sure!